### PR TITLE
Fixes #18258 - allow installing errata in batches

### PR DESCRIPTION
**Pull request by akofink**

### CODE REVIEW

File: app/models/setting/content.rb [hunk 1]
```diff
@@ -60,7 +60,9 @@ def self.load_defaults
                                               "such as virtual machines and DNS records may also be deleted."),
                  false, N_('Delete Host upon unregister')),
         self.set('register_hostname_fact', N_("When registering a host via subscription-manager, force use the specified fact (in the form of 'fact.fact')"),
-                 '', N_('Subscription manager name registration fact'), nil)
+                 '', N_('Subscription manager name registration fact'), nil),
+        self.set('erratum_install_batch_size', N_("Force pulp to install errata in batches. Set to -1 to disable."),
```
Comment (dLobatog) [inline app/models/setting/content.rb]: 0 would disable it according to the if condition above

File: app/lib/actions/katello/host/erratum/install.rb [hunk 2]
```diff
@@ -9,10 +9,20 @@ def plan(host, errata_ids)
             Type! host, ::Host::Managed
 
             action_subject(host, :errata => errata_ids)
-            plan_action(Pulp::Consumer::ContentInstall,
-                        consumer_uuid: host.content_facet.uuid,
-                        type:          'erratum',
-                        args:          errata_ids)
+            # Batch install errata in 6.2.z for BZ1288152
+            if Setting['erratum_install_batch_size'] && Setting['erratum_install_batch_size'] > 0
+              errata_ids.each_slice(Setting['erratum_install_batch_size']) do |errata_ids_batch|
```
Comment (dLobatog) [inline app/lib/actions/katello/host/erratum/install.rb]: How does this look in the UI when it's installed on batches? Just the last task?

Comment (akofink) [inline app/lib/actions/katello/host/erratum/install.rb]: @dLobatog It looks the same in the foreman_tasks UI, but you can notice the difference in Dynflow. Here's 40 errata installed in 2 batches:

image

File: app/models/setting/content.rb (hunk 1 above)
Comment (akofink) [inline app/models/setting/content.rb]: Indeed. Do you think I should change this to "Set to 0 to disable." ? It doesn't really matter to me.

**Comment by akofink** [comment]
@dLobatog I will definitely add tests. I wanted to get a preliminary review before putting in the effort to make sure I was on the right track. Thanks for the quick turnaround!

File: app/models/setting/content.rb (hunk 1 above)
Comment (dLobatog) [inline app/models/setting/content.rb]: `Set to 0 to install all errata in one batch` sounds more self-explanatory

File: app/models/setting/content.rb [hunk 3]
```diff
@@ -60,7 +60,9 @@ def self.load_defaults
                                               "such as virtual machines and DNS records may also be deleted."),
                  false, N_('Delete Host upon unregister')),
         self.set('register_hostname_fact', N_("When registering a host via subscription-manager, force use the specified fact (in the form of 'fact.fact')"),
-                 '', N_('Subscription manager name registration fact'), nil)
+                 '', N_('Subscription manager name registration fact'), nil),
+        self.set('erratum_install_batch_size', N_("Force pulp to install errata in batches. Set to 0 to install all errata in one batch."),
```
Comment (jlsherrill) [inline app/models/setting/content.rb]: I might reword this to "Errata installed via katello-agent will be triggered in batches of this size".  No real need to mention pulp, but its important to mention that its katello-agent (versus remote-execution)

Comment (jlsherrill) [inline app/models/setting/content.rb]: I'm okay with this PR, but the cause of the PR was to get around an issue with pulp timing out during large errata installs.  This is no longer an issue with katello 3.1 or 3.2 or 3.3.  So I think we should default it to '0' here.

File: app/lib/actions/katello/host/erratum/install.rb [hunk 4]
```diff
@@ -9,10 +9,19 @@ def plan(host, errata_ids)
             Type! host, ::Host::Managed
 
             action_subject(host, :errata => errata_ids)
-            plan_action(Pulp::Consumer::ContentInstall,
-                        consumer_uuid: host.content_facet.uuid,
-                        type:          'erratum',
-                        args:          errata_ids)
+            if Setting['erratum_install_batch_size'] && Setting['erratum_install_batch_size'] > 0
```
Comment (anandkumaragrawal) [inline app/lib/actions/katello/host/erratum/install.rb]: @akofink: How will these differ
if Setting['erratum_install_batch_size'] && Setting['erratum_install_batch_size'] > 0
& 
if Setting['erratum_install_batch_size'] > 0

Comment (akofink) [inline app/lib/actions/katello/host/erratum/install.rb]: @anandkumaragrawal The difference is seen when `Setting['erratum_install_batch_size']` is nil. Without the first check, you would get an `undefined method '>' for nil:NilClass` error.

Comment (anandkumaragrawal) [inline app/lib/actions/katello/host/erratum/install.rb]: @akofink thanks for the information, I would definitely add this to my knowledge.

Though, I would have kept this as below:
if Setting['erratum_install_batch_size'].to_i > 0

Comment (akofink) [inline app/lib/actions/katello/host/erratum/install.rb]: @anandkumaragrawal That would also certainly work for this case. Mostly this is an idiom in Ruby that many people follow. i.e. `if x && x.whatever`

Comment (anandkumaragrawal) [inline app/lib/actions/katello/host/erratum/install.rb]: Thanks @akofink for the information, will certainly help
